src/utils/withSuggestions.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+import React, { Component } from 'react'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import get from 'lodash/get'
+import SuggestionsList from '../components/Navigation/Ui/Suggestions/SuggestionsList'
+
+/**
+ * HoC that manage the suggestion functionality for a Component
+ * It retrieves suggested data for the specified field and show those into a selectable list
+ */
+function withSuggestions(WrappedComponent) {
+  class EnhanceSuggestions extends Component {
+    constructor(props) {
+      super(props)
+      this.cmp = React.createRef()
+    }
+
+    //Once a suggestion is applied, then it will added as a change for the current field
+    applySuggestion = suggestion => {
+      this.cmp.current.onChange(suggestion)
+    }
+
+    render() {
+      const { suggestedData, field } = this.props
+      return (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+          <WrappedComponent {...this.props} ref={this.cmp} />
+          {suggestedData && <SuggestionsList field={field} items={suggestedData} onSelect={this.applySuggestion} />}
+        </div>
+      )
+    }
+  }
+
+  return connect(
```

Comment:
Is not only the "connection" that identify what is a container and what is not, a containers doesn't have to be connected.
In general, it's a kind of components that contains logics, passed down to sub-components.
Also components, or in this case HoC, could be connected directly to the store, without to be considered containers.
In some cases it could be cleaner and smart connect a component to the store, instead of passing down tons of props, while those are mainly stored into the state of the application.

In the case of HoC, it ends up to be simpler using the component as-it-is instead of understand what kind of props it needs and passing them down each time.